docs/large-tutorials/HD143006_part_2.py

```diff
@@ -0,0 +1,480 @@
+    residuals()
+
+    fig, ax = plt.subplots(ncols=2, nrows=2, figsize=(10, 10))
+    im = ax[0, 0].imshow(
+        np.squeeze(model.icube.sky_cube.detach().cpu().numpy()),
+        origin="lower",
+        interpolation="none",
+        extent=model.icube.coords.img_ext,
+    )
+    plt.colorbar(im, ax=ax[0, 0])
+
+    im = ax[0, 1].imshow(
+        np.squeeze(residuals.sky_cube.detach().cpu().numpy()),
+        origin="lower",
+        interpolation="none",
+        extent=residuals.coords.img_ext,
+    )
+    plt.colorbar(im, ax=ax[0, 1])
+
+    im = ax[1, 0].imshow(
+        np.squeeze(torch.log(model.fcube.ground_amp.detach()).cpu().numpy()),
+        origin="lower",
+        interpolation="none",
+        extent=residuals.coords.vis_ext,
+    )
+    plt.colorbar(im, ax=ax[1, 0])
+
+    im = ax[1, 1].imshow(
+        np.squeeze(torch.log(residuals.ground_amp.detach()).cpu().numpy()),
+        origin="lower",
+        interpolation="none",
+        extent=residuals.coords.vis_ext,
+    )
+    plt.colorbar(im, ax=ax[1, 1])
+
+    return fig
+
+
+# ## Training loop
+
+# Now lets encapsulate the training loop into a function which we can easily re-run with different configuration values.
+#
+# To learn more information about the components of this training loop, please see the [Losses API](api.html#module-mpol.losses).
+
+
+# +
+from mpol import losses, connectors
+
+
+def train(model, dataset, optimizer, config, writer=None, logevery=50):
+    model.train()
+    residuals = connectors.GriddedResidualConnector(model.fcube, dataset)
+    for iteration in range(config["epochs"]):
+        optimizer.zero_grad()
+        vis = model.forward()  # calculate the predicted model
+        sky_cube = model.icube.sky_cube
+
+        loss = (
+            losses.nll_gridded(vis, dataset)
+            + config["lambda_sparsity"] * losses.sparsity(sky_cube)
+            + config["lambda_TV"] * losses.TV_image(sky_cube)
+            + config["entropy"] * losses.entropy(sky_cube, config["prior_intensity"])
+        )
+
+        if (
+            iteration % logevery == 0
+        ) and writer is not None:  # logging the loss and image for visualization and analysis
+            writer.add_scalar("loss", loss.item(), iteration)
+            writer.add_figure("image", log_figure(model, residuals), iteration)
+
+        loss.backward()  # calculate gradient of the parameters
+        optimizer.step()  # update the model parameters
+
+    return loss.item()
+
+
+# -
+
+# Now lets initialize the model to the pseudo-dirty image, set our hyperparameters, and create our optimizer.
+
+model.load_state_dict(
+    torch.load("model.pt")
+)  # load our initialized model from the previous section
+
+# Here is where we define the hyperparameters under the `config` dictionary. Most of these hyperparameters, such as `lambda_TV` and `entropy`, correspond to the scalar prefactors $\lambda$ as described in the [intro to RML](rml_intro.html).
+
+config = (
```

Comment:
Where do these config values come from? The text doesn't provide any context to why they might be set to the values that they are.